docs/design/software_architecture/index.md

```diff
@@ -4,59 +4,54 @@ This section explains an overview of software architecture.
 
 ![architecture](../../imgs/architecture.drawio.png)
 
-CARET is divided into three phases; recording phase, configuration phase and analyzing phase.
+CARET serves three phases; recording, configuration and visualization.
 
-## Recording Phase
+## Recording
 
-In the recording phase, CARET records system execution data from tracepoints.
+In the recording phase, CARET records events data obtained from tracepoints during application's runtime.
 
-In addition to the tracepoints built into ROS2, CARET adds tracepoints by several methods; CARET_trace CARET_rclcpp and TILDE.
-For flexibility in adding tracepoints, tracepoints are added by hooks as possible.
-Only if it is not possible to add tracepoints by hooks, tracepoints are added by other methods as a supplementary method.
+CARET adopts LTTng as a tracing mechanism. LTTng session daemon collects events from tracepoints embedded in a target application.
+`rclcpp` provided by ROS 2 has original tracepoints and CARET uses some of them also. CARET collects its dedicated tracepoints added by function hooking.
+To add tracepoints flexibly, function hooking is .
+Only if it is not possible to add tracepoints by function hooking due to some constraints from implementation, tracepoints are added via another approach.
 
-All recorded data is stored as TraceData and used in the analyzing phase.
+All of recorded events are stored to a set of trace data CTF-based. It is visualized for user to observe application's performance and behavior.
 
-CARET utilizes LTTng as a trace mechanism.
-
-In CARET, packages related to recording phase are followings.
-
-- [caret_trace](./caret_trace.md)
-- [TILDE](./tilde.md)
+[`caret_trace`](./caret_trace.md) is the main package for realizing recording. `caret_trace` collects events invoked in `rclcpp`, `rcl` and DDS. It is inconvenient to see actual time when data is consumed because data is consumed in a user code actually. [TILDE](./tilde.md) serves tracepoints to collect events which happens in a user code. CARET is able to refer to them for diving into events in a user code.
 
 See also
 
 - [Tracepoints](../trace_points/index.md)
 - [Runtime Processing](../runtime_processing/index.md)
 - [The LTTng Documentation](https://lttng.org/docs/)
 
-## Configuration Phase
-
-In the configuration phase, CARET needs configurations to calculate latency.
+## Configuration
 
-Some of the configurations can be defined via the Python API.
-Configuration can be saved as a YAML file (Architecture file) and edited.
-The architecture file is used repeatedly in the analyzing phase.
+In the configuration, CARET expects users to complement structure definitions of a target application before visualizing data. Politely speaking, CARET expects users to define intra-node data paths and inter-node data paths to calculate response time.
+Extracting such definitions mechanically is difficult because this step requires structure of a target application.
```

Comment:
```suggestion
It is difficult to extracting such definitions mechanically fom trace data because this step depends on implementation of a target application which is not able to grasp from ros-layer.
```

Reply:
I think that grammar on the suggested sentence is not good, but I'd like to change it with seeing your feedback.

Reply:
I think that this sentence should be more abstract due to the following reason.
CARET ask users to define inter-path data path because CARET does not know what to observe at first. Users has to define intra-node data path because CARET want to complement information to grasp relationship between sub and pub, which you pointed out.

There are a few reasons for defining paths and strict explanation will confuse users.

Reply:
I see. LGTM